zowe/vscode-extension-for-zowe
# Allow users to open data sets and USS files in another codepage

## Proposed changes

⚠️ This PR depends on #2634 - merge that one first, then update base branch of this PR

* Resolves #2435
* Fixes #2576
* Also backported `ZoweLocalStorage` class from "next" branch to store recent codepage history

## How to Test

* Test opening data set & USS file in default EBCDIC encoding
* Test opening data set & USS file in binary mode
* Test opening data set & USS file in custom encoding
  * In same session, close and re-open them - custom encoding should be remembered
  * Test with encoding in z/OSMF profile - custom encoding should take precedence
* Test saving USS file tagged as ISO8859-1 - USS tag should not change

## Release Notes




Milestone: 2.14.0

Changelog:
* Added "Open with Encoding" feature to open data sets and USS files in a non-standard codepage.
* Fixed issue where USS file tag could get overwritten when changes to file are uploaded.

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_


## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_


## Further comments



<img width="610" alt="image" src="https://github.com/zowe/vscode-extension-for-zowe/assets/22344007/c33fa6b6-0d42-4cdb-8a59-b4561bf34d6a">

https://github.com/zowe/vscode-extension-for-zowe/assets/22344007/d8cb4fb3-3d69-4ff1-8071-dedd318b3185
